packages/vitest-pool-workers/src/pool/index.ts

```diff
@@ -399,25 +338,41 @@ function buildProjectWorkerOptions(
 	// compatibility flags enabled. Vitest makes heavy use of Node APIs, and many
 	// of the libraries it depends on expect `require()` to return
 	// `module.exports` directly, rather than `{ default: module.exports }`.
+	// Initialize compatibilityFlags if undefined
 	runnerWorker.compatibilityFlags ??= [];
-	assertCompatibilityFlagEnabled({
-		compatibilityFlags: runnerWorker.compatibilityFlags,
-		compatibilityDate: runnerWorker.compatibilityDate,
-		relativeProjectPath: project.relativePath,
-		relativeWranglerConfigPath,
-		// https://developers.cloudflare.com/workers/configuration/compatibility-dates/#commonjs-modules-do-not-export-a-module-namespace
-		enableFlag: "export_commonjs_default",
-		disableFlag: "export_commonjs_namespace",
-		defaultOnDate: "2022-10-31",
-	});
-	assertCompatibilityFlagEnabled({
+
+	const flagAssertions = new FlagAssertions({
 		compatibilityFlags: runnerWorker.compatibilityFlags,
-		compatibilityDate: runnerWorker.compatibilityDate,
-		relativeProjectPath: project.relativePath,
+		optionsPath: `${OPTIONS_PATH}.miniflare`,
+		relativeProjectPath: project.relativePath.toString(),
 		relativeWranglerConfigPath,
-		enableFlag: "nodejs_compat",
 	});
 
+	const assertions = [
+		() =>
+			flagAssertions.assertDisableFlagNotPresent("export_commonjs_namespace"),
+		() =>
+			flagAssertions.assertEnableFlagOrCompatibilityDate(
+				"export_commonjs_default",
+				{
+					compatibilityDate: runnerWorker.compatibilityDate,
+					defaultOnDate: "2022-10-31",
+				}
+			),
+		() =>
+			flagAssertions.assertUnionOfEnableFlags([
+				"nodejs_compat",
+				"nodejs_compat_v2",
+			]),
+	];
+
+	for (const assertion of assertions) {
+		const result = assertion();
+		if (!result.isValid) {
+			throw new Error(result.errorMessage);
+		}
+	}
+
```

Comment:
Miniflare already exposes a function for working out what the current node_compat mode we are in (it is not trivial due to the negation flags).

See https://github.com/cloudflare/workers-sdk/blob/c1022d66c6536c9e8199ebffe73d2358aa4064ee/packages/miniflare/src/plugins/core/node-compat.ts#L24

I think that `isFlagPresent()` is actually too simple to cover the cases.

I would argue that we should just use the `getNodeCompat()` function from Miniflare, except that I see that we also need this other `export_commonjs_namespace` compat flag, which potentially also has complexity due to compat date default and negation.

I propose that if we are going to have this logic here, we just create a function called `isEnabled(enableFlag: string, disableFlag: string, defaultOnDate?: string)`, then you could just do:

```ts
if (!compatFlagAssertions.isEnabled("export_commonjs_default", "export_commonjs_namespace", "2022-10-31")) {
  throw new Error("export_commandjs_default must be enabled - either explicitly or by compat date being after 2022-10-31");
}
if (
  !compatFlagAssertions.isEnabled("nodejs_compat", "no_nodejs_compat") &&
  !compatFlagAssertions.isEnabled("nodejs_compat_v2", "no_nodejs_compat_v2")
  ) {
  throw new Error("Node.js compatibility must be enabled - by adding either `nodejs_compat` or `nodejs_compat_v2` compat flag");
}
```

Reply:
At the moment, we also have messaging depending on the options path, the wrangler path, and the relative project path. Are you suggesting we just drop that for a simpler set of messages? Most of the code in this class is dealing with messaging, so being liberated from that would simplify things significantly.

Reply:
OK looking at the code again, I think we could just reuse most of the logic in `assertCompatibiltyFlagEnabled()` but update it so that `enableFlag: string` becomes `enableFlags: string[]` and similarly with `disableFlag: string` becoming `disableFlags: string[]`.

The logic just needs a bit of tweaking to ensure that none of the `disableFlags` are in there and either at least one of the enableFlags is there or the compatibility date is new enough.

So then still throwing inside that function.

Reply:
And if you still want to move it out into a class that is cool too!

Reply:
I've updated this to consolidate the `assertDisableFlagNotPresent` and `assertEnableFlagOrCompatibilityDate` methods, but will wait for discussion on the messaging before proceeding further with the refactor.

Reply:
Sorry Pete, I didn't see your comment before pushing my latest update.